src/common/ServerCompatBanner.js

```diff
@@ -0,0 +1,101 @@
+/* @flow strict-local */
+
+import React from 'react';
+import { View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import store from '../boot/store';
+import { createStyleSheet, HALF_COLOR } from '../styles';
+import { useSelector, useDispatch } from '../react-redux';
+import Label from './Label';
+import { getActiveAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getSession } from '../directSelectors';
+import { dismissCompatNotice } from '../session/sessionActions';
+import ZulipTextButton from './ZulipTextButton';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+const styles = createStyleSheet({
+  wrapper: {
+    backgroundColor: HALF_COLOR,
+    padding: 8,
+  },
+  textRow: {
+    padding: 8,
+    flexDirection: 'row',
+  },
+  buttonsRow: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+});
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying that the server version is unsupported.
```

Comment:
```suggestion
 * A "nag banner" saying that the server version is unsupported, if so.
```

Otherwise it looks like this is something that needs to guarded by a conditional :slightly_smiling_face: